<center>
<h3>Universidade Federal do Rio Grande do Sul (UFRGS)<br>
Programa de Pós-Graduação em Engenharia Civil (PPGEC)<br>
PEC00144 - Métodos Experimentais em Engenharia Civil</h3>

<h1>Trabalho 2</h1>

<h3>Aluno: <a href="https://github.com/dutitello/">Eduarto Pagnussat Titello</a></h3>

</center>




Este trabalho tem por objetivo: no tema de interesse, adotar um problema imaginário de formulação explícita, com 2 ou mais parametros dimensionais. Desenhar, formular e definir uma nova escala para o problema, fazendo uma análise de mudança de escala de forma a poder construír tal modelo.

In [1]:
# Importando e configurando módulos
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt
#import matplotlib
%config InlineBackend.figure_format = 'svg' # Muda backend do jupyter para SVG ;)
import jupyter2latex as j2l # Uma maneira que encontrei para tabelas ficarem ok (github.com/dutitello/Jupyter2Latex)
from math import pi


# Apresentação do problema

O problema adotado consiste na avaliação das duas primeiras frequências naturais de vibração de um poste em concreto armado, considerando seu comportamento como elástico. O poste adotado tem comprimento efetivo $L= 12m$, seção constante de diâmetro externo $d_e = 50cm$ e espessura $e=6 cm$, resultando em um diâmetro interno $d_i = 38cm$, conforme a figura:

![Poste](./Images/Poste.png)

Conforme Rocha (2020a), as frequências naturais de vibracão de uma viga engastada livre (modelo adotado para o poste) podem ser obtidas por \ref{eq:wn}:

\begin{equation}
f_n = \frac{1}{2\pi} {\Big( \frac{\alpha_n}{L} \Big)}^2 \sqrt{\frac{EI}{m}}
\label{eq:wn}
\end{equation}

onde, para as duas primeiras frequências naturais, $\alpha_1=1.88$ e $\alpha_2=4.69$, respectivamente. 

A área da seção transversal circular vazada $A$ e seu momento de inércia $I$ são dados por:

\begin{equation}
A = \frac{\pi}{4}\big(d_e^2 - d_i^2 \big)
\label{eq:Ac}
\end{equation}

\begin{equation}
I = \frac{\pi}{64}\big(d_e^4 - d_i^4 \big)
\label{eq:Ic}
\end{equation}



Adotando o subíndice "$_c$" para representar a estrutura real em concreto, sendo empregado concreto da classe C25, o módulo de elasticidade do material é $E_c=28 GPa$, enquanto sua densidade é aproximadamente $\rho_c=2500 kg/m^3$. As propriedades da estrutura original são então:

In [2]:
L_c   = 12.0   # m
de_c  = 0.50   # m
di_c  = 0.38   # m
E_c   = 28.0E9 # N/m^2
rho_c = 2500   # kg/m^3

A_c  = pi/4*(de_c**2 - di_c**2)  # m^2
I_c  = pi/64*(de_c**4 - di_c**4) # m^4

EI_c = E_c*I_c   # Nm^2
μL_c = A_c*rho_c # kg/m
m_c  = μL_c*L_c  # kg

print(f'''
Propriedades da estrutura:
    Área da seção         A_c = {A_c:.3f} m^2
    Momento de Inércia    I_c = {I_c:.3E} m^4 
    Rigidez EI         (EI)_c = {EI_c:.3E} N.m^2
    Massa por compr.     μL_c = {μL_c:.2f} kg/m
    Massa total           m_c = {m_c:.2f} kg
''')


Propriedades da estrutura:
    Área da seção         A_c = 0.083 m^2
    Momento de Inércia    I_c = 2.044E-03 m^4 
    Rigidez EI         (EI)_c = 5.724E+07 N.m^2
    Massa por compr.     μL_c = 207.35 kg/m
    Massa total           m_c = 2488.14 kg



# Definição do modelo reduzido

Conforme a expressão \ref{eq:wn}, as frequências naturais da estrutura são dependentes do seu comprimento $L$, da sua rigidez à flexão $EI$ e da sua massa $m$. Em relação às dimensões o fator de escala adotado será 1:25.Adotando um modelo reduzido em alumínio, com módulo de elasticidade $E_m=70GPa$ e densidade $\rho_m=2700kg/m^3$, os demais fatores de escala serão relativos à rigidez $EI$ e a densidade $\rho$. Em relação à densidade a escala do modelo é simplesmente 2700:2500. 

Quanto à rigidez $EI$, o fator de escala deve considerar perfis comercialmente viáveis, visto que $E_m$ é fixo, dessa forma é adotado o perfil tubular TR-004 da Aluita (Aluita, 2019), de diâmetro externo $d_{e,m}=12.70mm$ e diâmetro interno $d_{i,m}=10.22mm$. Assim:

In [3]:
de_m  = 12.70/1000 # m
di_m  = 10.22/1000 # m
E_m   = 70.0E9     # N/m^2
rho_m = 2700       # kg/m^3

I_m  = pi/64*(de_m**4 - di_m**4) # m^4
EI_m = E_m*I_m   # Nm^2

print(f'''
Propriedades do modelo reduzido:
    Momento de Inércia      I_m = {I_m:.3E} m^4 
    Rigidez EI           (EI)_m = {EI_m:.3E} N.m^2
''')


Propriedades do modelo reduzido:
    Momento de Inércia      I_m = 7.415E-10 m^4 
    Rigidez EI           (EI)_m = 5.190E+01 N.m^2



In [4]:
escala_L  = 1/25
escala_m  = rho_m/rho_c
escala_EI = EI_m/EI_c

print(f'''
Fatores de escala fornecidos:
    Comprimento   L = {escala_L:.3E} 
    Massa         m = {escala_m:.3E} 
    Rigidez      EI = {escala_EI:.3E}
''')


Fatores de escala fornecidos:
    Comprimento   L = 4.000E-02 
    Massa         m = 1.080E+00 
    Rigidez      EI = 9.067E-07



# Análise do modelo reduzido 

Para que os resultados obtidos no modelo reduzido possam ser utilizados esses devem ser escalados de forma a representarem a estrutura real, para isso os dimensionais que envolvem o problema devem ter seus fatores de escala determinados conforme as escalas impostas. Para obtenção de tais fatores de escala, a base fundamental do sistema de unidades deve corresponder às unidades das escalas impostas, dessa forma o sistema $L, M, T$ (comprimento, massa, tempo) é transformado em um sistema $L, M, EI$ (comprimento, massa, rigidez EI), denominado no código como ABC. Cabe observar que na base ABC a massa é representada por $m$ e não $M$. 

In [5]:
# Importando DimData 
DimData = pd.read_excel('../resources/DimData.xlsx', 
                         index_col  =  0,
                         sheet_name = 'DimData')


# Unidades fundamentais originais 
LMT = ['L', 'M', 'T']

# Novas unidades fundamentais
ABC = ['L', 'm', 'EI'] 

# Importa matriz dimensional de ABC na base LMT
base = DimData.loc[ABC, LMT]
j2l.df2table(base, 'L, M, EI na base L, M, T')

# Inverte base de unidades de LMT para ABC
base_i = pd.DataFrame(np.linalg.inv(base), index=LMT, columns=ABC)
j2l.df2table(base_i, 'L, M, T na base L, M, EI')

L, M, EI na base L, M, T 
 
 
 
 
 L 
 M 
 T 
 
 
 
 
 L 
 1 
 0 
 0 
 
 
 m 
 0 
 1 
 0 
 
 
 EI 
 3 
 1 
 -2

L, M, T na base L, M, EI 
 
 
 
 
 L 
 m 
 EI 
 
 
 
 
 L 
 1.0 
 0.0 
 0.0 
 
 
 M 
 0.0 
 1.0 
 0.0 
 
 
 T 
 1.5 
 0.5 
 -0.5

Conforme \ref{eq:wn}, os parametros dimensionais relacionados as frequencias naturais de vibração são $f$, $L$, $m$ e $EI$, sendo que desses apenas $f$ não foi imposto. Formando a matriz dimensional $\bf D$ do problema no sistema ABC temos que: 

In [6]:
par = ['f', 'L', 'm', 'EI']
npar = len(par)

DMat_LMT = DimData.loc[par, LMT]
DMat_ABC = np.matmul(DMat_LMT, base_i)
DMat_ABC.rename(columns=dict(zip(LMT, ABC)), inplace=True) # Renomeia colunas para nova base
j2l.df2table(DMat_ABC, 'Matriz D na base ABC')

Matriz D na base ABC 
 
 
 
 
 L 
 m 
 EI 
 
 
 
 
 f 
 -1.5 
 -0.5 
 0.5 
 
 
 L 
 1.0 
 0.0 
 0.0 
 
 
 m 
 0.0 
 1.0 
 0.0 
 
 
 EI 
 0.0 
 0.0 
 1.0

Através da matriz dimensional $\bf D$ no sistema ABC os parametros de escala podem ser cálculados:

In [10]:
# Ok, não é "bonito" isso de ficar fazendo escala=f(escala)...
escalas = np.array([escala_L, escala_m, escala_EI])
escalas = np.tile(escalas, (npar, 1))
escalas = np.prod(escalas**DMat_ABC, axis=1)
escalas = pd.DataFrame({'λ': λs, '1/λ': 1/λs}, index=par)
j2l.df2table(escalas, 'Fatores de escala')

Fatores de escala 
 
 
 
 
 λ 
 1/λ 
 
 
 
 
 f 
 1.145323e-01 
 8.731160e+00 
 
 
 L 
 4.000000e-02 
 2.500000e+01 
 
 
 m 
 1.080000e+00 
 9.259259e-01 
 
 
 EI 
 9.066920e-07 
 1.102910e+06

Assim o poste pode ter suas frequencias naturais de vibração determinadas experimentalmente por um modelo em alumínio com apenas 4% da sua altura, desde que a frequência obtida seja escala pelo fator 8.73. 